supabase/migrations/2023101016_remote_schema.sql

```diff
@@ -0,0 +1,317 @@
+    NO MAXVALUE
+    CACHE 1
+);
+
+CREATE TABLE IF NOT EXISTS "public"."models" (
+    "id" bigint NOT NULL,
+    "name" text,
+    "type" text,
+    "created_at" timestamp with time zone DEFAULT now() NOT NULL,
+    "user_id" uuid,
+    "status" text DEFAULT 'processing'::text NOT NULL,
+    "modelId" text
+);
+
+ALTER TABLE "public"."models" OWNER TO "postgres";
+
+ALTER TABLE "public"."models" ALTER COLUMN "id" ADD GENERATED BY DEFAULT AS IDENTITY (
+    SEQUENCE NAME "public"."models_id_seq"
+    START WITH 1
+    INCREMENT BY 1
+    NO MINVALUE
+    NO MAXVALUE
+    CACHE 1
+);
+
+CREATE TABLE IF NOT EXISTS "public"."samples" (
+    "id" bigint NOT NULL,
+    "uri" text NOT NULL,
+    "modelId" bigint NOT NULL,
+    "created_at" timestamp with time zone DEFAULT now() NOT NULL
+);
+
+ALTER TABLE "public"."samples" OWNER TO "postgres";
+
+ALTER TABLE "public"."samples" ALTER COLUMN "id" ADD GENERATED BY DEFAULT AS IDENTITY (
+    SEQUENCE NAME "public"."samples_id_seq"
+    START WITH 1
+    INCREMENT BY 1
+    NO MINVALUE
+    NO MAXVALUE
+    CACHE 1
+);
+
+-- New tables
+CREATE TABLE IF NOT EXISTS "public"."users" (
+    "id" uuid PRIMARY KEY,
+    "email" text,
+    "name" text,
+    "credit_balance" integer,
+    "last_credit_update" timestamp with time zone,
+    "created_at" timestamp with time zone DEFAULT now(),
+    "updated_at" timestamp with time zone DEFAULT now(),
+    "last_login_at" timestamp with time zone
+);
+
+CREATE TABLE IF NOT EXISTS "public"."plans" (
+    "id" bigint PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,
+    "name" text,
+    "description" text,
+    "price_in_usd" decimal(10,2),
+    "total_credits" integer,
+    "max_trainings" smallint,
+    "max_generations" smallint,
+    "max_edits" smallint,
+    "paddle_product_id" text,
+    "duration" smallint,
+    "billing_cycle" text,
+    "is_active" boolean,
+    "meta_data" text,
+    "created_at" timestamp with time zone DEFAULT now(),
+    "updated_at" timestamp with time zone DEFAULT now()
+);
+
+CREATE TABLE IF NOT EXISTS "public"."user_subscriptions" (
+    "id" bigint PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,
+    "user_id" uuid REFERENCES "public"."users"("id"),
+    "plan_id" bigint REFERENCES "public"."plans"("id"),
+    "remaining_training" integer,
+    "remaining_generation" integer,
+    "remaining_edits" integer,
+    "start_date" date,
+    "end_date" date,
+    "is_active" boolean,
+    "auto_renew" boolean,
+    "created_at" timestamp with time zone DEFAULT now(),
+    "updated_at" timestamp with time zone DEFAULT now()
+);
+
+CREATE TABLE IF NOT EXISTS "public"."subscription_history" (
+    "id" bigint PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,
+    "subscription_id" bigint REFERENCES "public"."user_subscriptions"("id"),
+    "start_date" date,
+    "end_date" date,
+    "status" text,
+    "change_type" text,
+    "created_at" timestamp with time zone DEFAULT now()
+);
+
+CREATE TABLE IF NOT EXISTS "public"."credit_transactions" (
```

Comment:
remove this table and add the new columns to the credits table and rename it then